### PR TITLE
move to v2 and print related nodes

### PR DESCRIPTION
**Pull request by badtant**
As suggested here: https://github.com/dylanb/react-axe/issues/10

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -91,13 +101,15 @@ function checkAndReport(node, timeout) {
...
 					console.groupCollapsed('%c%s: %c%s %s', fmt, result.impact, defaultReset, result.help, result.helpUrl);
 					result.nodes.forEach(function (node) {
-						var el = document.querySelector(node.target.toString());
-						console.error(node.failureSummary);
-						console.log('HTML: %c%s', boldCourier, node.html);
-						if (!el) {
-							console.log('Selector: %c%s', boldCourier, node.target.toString());
-						} else {
-							console.log('Element: %o', el);
+						console.error(node.any[0].message);
```
Comment (dylanb) [inline index.js]: This is not sufficient code to generate the summary correctly. Check out this code - which you may have to duplicate if it is not exposed via the `axe.commons` or `axe.utils` objects

https://github.com/dequelabs/axe-core/blob/master/lib/core/reporters/helpers/failure-summary.js

File: index.js [hunk 2]
```diff
@@ -91,13 +101,15 @@ function checkAndReport(node, timeout) {
...
+						logHtmlAndElement(node);
+
+						if (node.any[0].relatedNodes.length > 0) {
+							console.groupCollapsed('Related nodes');
+							node.any[0].relatedNodes.forEach(function (relatedNode) {
```
Comment (dylanb) [inline index.js]: This is also not sufficient. The above code should show you how to also find all the related nodes for all of the checks that fail the rule.

**Comment by badtant** [comment]
Cool, I've changed to the way I think you describe.
"failure summary" wasn't exposed so i copied it.

File: failure-summary.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+var failureSummary = function failureSummary(nodeData) {
+	'use strict';
+
+	var failingChecks = {};
+	// combine "all" and "none" as messaging is the same
+	failingChecks.none = nodeData.none.concat(nodeData.all);
+	failingChecks.any = nodeData.any;
+
+	return Object.keys(failingChecks).map(function (key) {
```
Comment (dylanb) [inline failure-summary.js]: There should be a message in front of each grouping to instruct the user how to interpret the output. In the case of `any` this message should be along the lines of "fix any of the following" and for the `none` group it should be "fix all of the following".

File: index.js [hunk 4]
```diff
@@ -91,13 +102,19 @@ function checkAndReport(node, timeout) {
...
-							console.log('Element: %o', el);
+						console.error(failureSummary(node));
+						logHtmlAndElement(node);
+
+						let allAnyNone = node.all.concat(node.any, node.none);
+
+						if (allAnyNone.length > 0) {
+							console.groupCollapsed('Related nodes');
+							allAnyNone.forEach(function (check) {
+								check.relatedNodes.forEach(function (relatedNode) {
```
Comment (dylanb) [inline index.js]: This is going to confuse users if multiple checks return related nodes. In that case, they will not be able to tell which related node belongs to which failed check. The related node information should be interspersed with the appropriate check messages. As you can see from this file https://github.com/dequelabs/axe-core/blob/master/lib/core/reporters/helpers/format-check.js each check is formatted on its own with its related nodes and then the entire thing is combined into the overall message for the node.

Comment (badtant) [inline index.js]: hopefully getting closer :)

**Comment by badtant** [comment]
hopefully getting closer :)

File: failure-summary.js (hunk 3 above)
Comment (dylanb) [inline failure-summary.js]: @badtant I saw fixes for all the other comments except this one

Comment (badtant) [inline failure-summary.js]: I call failureSummary for both any and none. And failureSummary prints the massage.

```
failureSummary(node, 'any');
failureSummary(node, 'none');
```

Comment (badtant) [inline failure-summary.js]: or am I missing something?

Comment (dylanb) [inline failure-summary.js]: @badtant where is the message that says "fix all of" and "fix one of"? I could not find it.

Comment (badtant) [inline failure-summary.js]: failureSummary is called for both "any" and "none". then failureSummary will call failureMessage which I copied from https://github.com/dequelabs/axe-core/blob/master/lib/core/reporters/helpers/failure-summary.js

thats where the message come from.

Comment (dylanb) [inline failure-summary.js]: ok, I see the message

Here is what it looks like in the example on my machine

image

The order of the messages is confusing. It should go:

Element (group)
->HTML
->Failure Summary (group)
->->Related nodes for failure summary

WDYT?

Comment (badtant) [inline failure-summary.js]: Have a look now.
Didn't like hiding away the related nodes under failure summary though.

File: index.js [hunk 5]
```diff
@@ -48,6 +48,42 @@ function getCommonParent(nodes) {
...
+		return check.message || '';
+	}));
+}
+
+function failureSummary(node, key) {
+	if (node[key].length > 0) {
+		logElement(node, console.groupCollapsed);
+			logHtml(node);
+			console.error(failureMessage(node, key));
+			console.groupCollapsed('Related nodes');
```
Comment (dylanb) [inline index.js]: This should be conditional, only if related nodes exist

File: failure-summary.js (hunk 3 above)
Comment (dylanb) [inline failure-summary.js]: It is definitely much better now, you just need to make the related nodes information conditional and we can merge. There is one condition that will not work but I can fix that once you merge.

Comment (badtant) [inline failure-summary.js]: fixed so related nodes information is conditional.
also added basic eslint and editorconfig. hope you don't mind.

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,8 @@
+root = true
+
+[*]
+indent_style = space
```
Comment (dylanb) [inline .editorconfig]: change this to tabs - that is what we were using before

Comment (badtant) [inline .editorconfig]: done

**Pull request merged**